pkg/exporter/exporter.go

```diff
@@ -164,6 +180,12 @@ func New(server string, timeout time.Duration, logger log.Logger) *Exporter {
 			nil,
 			nil,
 		),
+		rejectedConnections: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "", "rejected_connections_total"),
```

Comment:
This should be `memcached_connections_rejected_total` so it groups well with other `memcached_connections_*` metrics.

Reply:
That doesn't fit with Prometheus naming conventions. If we're going to add new metrics, we should follow the correct conventions going foward.

Reply:
https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L1188-L1323
I thought the naming is the same as protocol.txt, but it is wrong.

I will change memcached_rejected_connections_total to  memcached_connections_rejected_total.

BTW, there are `memcached_max_connections` and `memcached_current_connections` .
I'll leave these unchanged.

Reply:
@SuperQ Can you please refer to the written down conventions you seem to refer to here? I'm not sure what you're referring to here.

My suggestion is precisely to follow the existing style of ordering namespaces from least specific to most specific. So it's `<namespace>_<subsystem>_<metric>[_<unit>[_<suffix>]]` and `connections` is the logical subsystem for me here to group all metrics related to memcached connections. This is the pre-dominent style I see in the Prometheus server and which has been very helpful when discovering metrics as related metrics are automatically grouped together in lists.

Reply:
`<subsystem>` is optional, like `process_resident_memory_bytes`. The problem is `rejected` is not a countable, and therefore not a unit. `connections` is the countable unit in this case.

Reply:
Maybe we need to add an explanation of what "unit" means in the best practices, that it should be a countable thing.

Reply:
I don't see how the best practices page provides any clear guidance for either style here, or how `memcached_connections_rejected_total` wouldn't fit the documented best practices.

You seem to have a personal preference to follow a natural English style (for which there are examples like `prometheus_tsdb_out_of_order_exemplars_total`), where I prefer the ordered style (existing examples are `prometheus_tsdb_checkpoint_deletions_failed_total`, `prometheus_tsdb_wal_truncations_failed_total`). I'm reading the `connections_rejected` as `connections (rejected)` where the "rejected" simply further specifies the connection type. 

Looking at the Prometheus server metrics alone you'll find many examples for both styles. I'm not able to see any clear preference for either one. If you wish the project to adopt a certain one, I welcome you to start such discussion in the dev summits or the Prometheus documentation repo.

Reply:
Yes, some of those Prometheus metrics seem to be incorrectly named.

Some additional documentation/guideline is needed.

Reply:
I'm a bit surprised about your usage of terms like "incorrect" and "naming convention" as there is apparently no consensus on that.

I'll be happy to follow any stricter metric naming convention once it has materialized, but so far I haven't seen anything in that direction. FWIW, at this point I'd also vote against your proposal presented here as I fail to see how that's an improvement.

Reply:
I filed https://github.com/prometheus/prometheus/issues/8718 to see if we can clarify the unit vs unit modifier in either the metrics or the best practices docs.